File: Python3/0303. Range Sum Query - Immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.temp = [0]
        for i in nums:
            self.temp.append(self.temp[-1]+i)

    def sumRange(self, i: int, j: int) -> int:
        return self.temp[j+1]-self.temp[i]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)

File: C++/0303. Range Sum Query - Immutable.cpp
#include<vector>
class NumArray {
public:
    vector<int> sums={0};
    NumArray(vector<int>& nums) {
        for(auto n:nums){
            sums.push_back(sums.back()+n);
        }
    }
    
    int sumRange(int i, int j) {
        return sums[j+1] - sums[i];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(i,j);
 */
